SerialPrograms/Source/PokemonSwSh/Inference/PokemonSwSh_BoxNatureDetector.cpp

```diff
@@ -0,0 +1,101 @@
+/*  IV Checker Reader
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "Common/Cpp/Color.h"
+#include "CommonFramework/GlobalSettingsPanel.h"
+#include "CommonFramework/ImageTools/ImageBoxes.h"
+#include "CommonFramework/ImageTools/ImageFilter.h"
+#include "CommonFramework/ImageTypes/ImageRGB32.h"
+#include "CommonFramework/ImageTools/ImageStats.h"
+#include "CommonFramework/ImageTypes/ImageViewRGB32.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "PokemonSwSh_BoxNatureDetector.h"
+
+#include <iostream>
+using std::cout;
+using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSwSh{
+
+BoxNatureDetector::BoxNatureDetector(VideoOverlay& overlay)
+    : m_box_atk(overlay, { 0.689, 0.198 + 1 * 0.0515, 0.84, 0.047})
+    , m_box_def(overlay, { 0.689, 0.198 + 2 * 0.0515, 0.84, 0.047})
+    , m_box_spatk(overlay, { 0.689, 0.198 + 3 * 0.0515, 0.84, 0.047})
+    , m_box_spdef(overlay, { 0.689, 0.198 + 4 * 0.0515, 0.84, 0.047})
+    , m_box_spd(overlay, { 0.689, 0.198 + 5 * 0.0515, 0.84, 0.047})
+{}
+
+NaturePlusMinus BoxNatureDetector::read(Logger& logger, const ImageViewRGB32& frame, const OverlayBoxScope& box){
+
+    const auto region = extract_box_reference(frame, box);
+
+    // Retain only red pixels from region
+    const bool replace_color_within_range = false;
+    const ImageRGB32 red_region = filter_rgb32_range(
+        region,
+        combine_rgb(150, 0, 0), combine_rgb(255, 100, 100), Color(0), replace_color_within_range
+    );
+    const size_t num_red_pixels = image_stats(red_region).count;
+
+    // Retain only blue pixels from region
+    const ImageRGB32 blue_region = filter_rgb32_range(
+        region,
+        combine_rgb(0, 0, 180), combine_rgb(130, 130, 255), Color(0), replace_color_within_range
```

Comment:
Can you check to make sure this works on both of these test images with room to spare on the thresholds? These are two different capture cards, one of which is has significant color distortion.
![screenshot-20230128-010949517344](https://user-images.githubusercontent.com/4991305/215257751-36141a4f-0720-4a36-8b7f-93b1a22c3b7c.png)

Probably want to ping Gin later on to add these to the test unit tests.

![screenshot-20230128-010921021655](https://user-images.githubusercontent.com/4991305/215257746-4b5df19e-8a4d-48b6-8211-ef7083e25db8.png)